provider/datagen/src/source.rs

```diff
@@ -110,44 +119,55 @@ impl SourceData {
         if tag == "release-71-1" {
             tag = "icu4x/2022-08-17/71.x";
         }
-        self.with_icuexport(
-            cached_path::CacheBuilder::new().freshness_lifetime(u64::MAX).build().and_then(|cache| cache
-                .cached_path(
-                    &format!("https://github.com/unicode-org/icu/releases/download/{}/icuexportdata_{}.zip", tag, tag.replace('/', "-"))
-            )).map_err(|e| DataError::custom("Download").with_display_context(&e))?)
+        let resource = format!(
+            "https://github.com/unicode-org/icu/releases/download/{}/icuexportdata_{}.zip",
+            tag,
+            tag.replace('/', "-")
+        );
+        Ok(Self {
+            icuexport_paths: Some(Arc::new(SerdeCache::new(AbstractFs::new_lazy(Box::new(
+                move || download(&resource),
+            ))))),
+            ..self
+        })
     }
 
     /// Adds CLDR data to this `DataSource`. This data will be downloaded from the `latest` GitHub tag.
     pub fn with_cldr_latest(self, locale_subset: CldrLocaleSubset) -> Result<Self, DataError> {
-        let response = reqwest::blocking::Client::new()
-            .head("https://github.com/unicode-org/cldr-json/releases/latest")
-            .send()
-            .map_err(|e| DataError::custom("reqwest error").with_display_context(&e))?;
-        self.with_cldr_for_tag(
-            response
-                .url()
-                .path()
-                .split('/')
-                .next_back()
-                .expect("split is non-empty"),
-            locale_subset,
-        )
+        let path = Box::new(move || {
+            let tag = follow_redirect_and_get_last_element(
+                "https://github.com/unicode-org/cldr-json/releases/latest",
+            )?;
```

Comment:
Issue (pre-existing): it's extremely brittle to use the _actual_ latest release here. We only guarantee that `icu4x-datagen` is compatible with the current version of CLDR, and all past versions of CLDR since 1.0. However, every release of CLDR is essentially a breaking change (this is not the case for ICU); code that runs against CLDR JSON 42 may not work on CLDR JSON 43.

I suggest hard-coding the _latest supported_ CLDR tag in the datagen code somewhere, which we should update whenever we turn the crank on integrating a new CLDR version.

Reply:
I guess clients will see an error then and download a newer binary?

I wouldn't want to stop anyone from using a newer CLDR, we could print a warning that it's not been tested with this version, but it's likely going to work.

Reply:
Assuming clients will download a newer binary is an unreliable solution:

1. CLDR might make a release before ICU4X has had time to guarantee compatibility
2. We might stop updating 1.x binaries when 2.x is available
3. Clients might not think to upgrade their binary if it suddenly stops working